inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TimestampAddFunction.java

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Function;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+/**
+ * TimestampAddFunction
+ * Description: Add integer expression intervals to the date or date time expression expr.
+ *  The unit of the time interval is specified by the unit parameter, which should be one of the following values:
+ *  FRAC_SECOND, SECOND, MINUTE, HOUR, DAY, WEEK, MONTH, QUARTER, or YEAR.
+ */
+public class TimestampAddFunction implements ValueParser {
+
+    private ValueParser intervalParser;
+    private ValueParser amountParser;
+    private ValueParser datetimeParser;
+    private static final String DEFAULT_FORMAT_DATE_TIME = "yyyy-MM-dd HH:mm:ss";
+    private static final String DEFAULT_FORMAT_DATE = "yyyy-MM-dd";
+
+    public TimestampAddFunction(Function expr) {
+        List<Expression> expressions = expr.getParameters().getExpressions();
+        intervalParser = OperatorTools.buildParser(expressions.get(0));
+        amountParser = OperatorTools.buildParser(expressions.get(1));
+        datetimeParser = OperatorTools.buildParser(expressions.get(2));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        String interval = intervalParser.parse(sourceData, rowIndex, context).toString();
+        Long amount = Long.parseLong(amountParser.parse(sourceData, rowIndex, context).toString());
+        String dateString = datetimeParser.parse(sourceData, rowIndex, context).toString();
+        return evalDate(dateString, interval, amount);
+    }
+
+    private String evalDate(String dateString, String interval, Long amount) {
+        DateTimeFormatter formatter = null;
+        LocalDateTime dateTime = null;
+        boolean hasTime = true;
+        if (dateString.indexOf(' ') != -1) {
+            formatter = DateTimeFormatter.ofPattern(DEFAULT_FORMAT_DATE_TIME);
+            dateTime = LocalDateTime.parse(dateString, formatter);
+        } else {
+            formatter = DateTimeFormatter.ofPattern(DEFAULT_FORMAT_DATE);
+            dateTime = LocalDate.parse(dateString, formatter).atStartOfDay();
+            hasTime = false;
+        }
+
+        switch (interval.toUpperCase()) {
+            case "FRAC_SECOND":
```

Comment:
why not support microsecond and millisecond?

Reply:
"FRAC-SECOND" stands for milliseconds. No matter how high the accuracy is, other databases no longer support it. There is an error here that has been corrected.